include/swift/AST/Decl.h

```diff
@@ -32,6 +32,7 @@
 #include "llvm/Support/TrailingObjects.h"
 
 namespace swift {
+  class AccessScope;
```

Comment:
Since we're returning by value, a forward-declaration isn't good enough. The header is cheap enough to just include, I think.

(Unless there's a circularity issue that I've forgotten, in which case we should figure out a better way to fix it.)